convert2rhel/subscription.py

```diff
@@ -244,11 +253,33 @@ def install_rhel_subscription_manager():
     )
     if ret_code:
         loggerinst.critical("Failed to install subscription-manager packages. See the above yum output for details.")
-    else:
-        loggerinst.info("Packages installed:\n%s" % "\n".join(rpms_to_install))
-        pkg_names = get_installed_submgr_pkg_names(rpms_to_install)
-        utils.changed_pkgs_control.track_installed_pkgs(pkg_names)
-        loggerinst.debug("Tracking installed packages: %r" % pkg_names)
+
+    loggerinst.info("\nPackages installed:\n%s" % "\n".join(rpms_to_install))
+
+    # Untrack any packages that are present in the system before the install
+    if rpms_to_untrack and pkg_names:
+        for rpm in rpms_to_untrack:
+            loggerinst.debug("Untracking package: %s" % rpm)
+            pkg_names.remove(rpm)
+
+    loggerinst.debug("Tracking installed packages: %r" % pkg_names)
+    utils.changed_pkgs_control.track_installed_pkgs(pkg_names)
+
+
+def check_for_pkg_presence(pkg_names):
```

Comment:
1. The function name `check_for_pkg_presence` does not imply clearly what it does. I'd rename it to e.g. `filter_installed_pkgs()`.
2. The function would be better suited for `pkghandler.py` rather than this `subscription.py` module.
3. The line `_, ret_code = utils.run_subprocess(["rpm", "-q", pkg], print_output=False)` can be moved to a separate function `is_pkg_installed()`
    - this new `is_pkg_installed()` can be used instead of `pkghandler.get_installed_pkg_objects()` within `subscription.py` where simple true/false would suffice (i.e. no need to call yum/dnf python API to retrieve pkg objects)